diff --git a/gen-src/main/java/org/projectfloodlight/openflow/protocol/ver13/OFActionIdVer13.java b/gen-src/main/java/org/projectfloodlight/openflow/protocol/ver13/OFActionIdVer13.java
index fe11d66..20990d6 100644
--- a/gen-src/main/java/org/projectfloodlight/openflow/protocol/ver13/OFActionIdVer13.java
+++ b/gen-src/main/java/org/projectfloodlight/openflow/protocol/ver13/OFActionIdVer13.java
@@ -46,8 +46,14 @@ abstract class OFActionIdVer13 {
             bb.readerIndex(start);
             switch(type) {
                case (short) 0xffff:
-                   // discriminator value OFActionType.EXPERIMENTER=65535 for class OFActionIdExperimenterCutVer13
-                   return OFActionIdExperimenterCutVer13.READER.readFrom(bb);
+                   try {
+                       // discriminator value OFActionType.EXPERIMENTER=65535 for class OFActionIdExperimenterCutVer13
+                       return OFActionIdExperimenterCutVer13.READER.readFrom(bb);
+                   } catch (OFParseError exception) {
+                       bb.readerIndex(start);
+                       // discriminator value OFActionType.EXPERIMENTER=65535 for class OFActionIdExperimenterCutVer13
+                       return OFActionIdExperimenterVer13.READER.readFrom(bb);
+                   }
                case (short) 0xc:
                    // discriminator value OFActionType.COPY_TTL_IN=12 for class OFActionIdCopyTtlInVer13
                    return OFActionIdCopyTtlInVer13.READER.readFrom(bb);
