diff --git a/Pcap++/header/DpdkDeviceList.h b/Pcap++/header/DpdkDeviceList.h
index f5eb5fd..9b66d9e 100644
--- a/Pcap++/header/DpdkDeviceList.h
+++ b/Pcap++/header/DpdkDeviceList.h
@@ -127,6 +127,11 @@ namespace pcpp
 		 */
 		static bool initDpdk(CoreMask coreMask, uint32_t mBufPoolSizePerDevice, uint8_t masterCore = 0);
 
+         /**
+         * Only set internal variables, used for advances dpdk eal init
+         */
+         static bool externalInitializationDpdk(CoreMask coreMask, uint32_t mBufPoolSizePerDevice, uint8_t masterCore = 0);
+
 		/**
 		 * Get a DpdkDevice by port ID
 		 * @param[in] portId The port ID
diff --git a/Pcap++/src/DpdkDeviceList.cpp b/Pcap++/src/DpdkDeviceList.cpp
index bf0522f..50348d2 100644
--- a/Pcap++/src/DpdkDeviceList.cpp
+++ b/Pcap++/src/DpdkDeviceList.cpp
@@ -145,6 +145,15 @@ bool DpdkDeviceList::initDpdk(CoreMask coreMask, uint32_t mBufPoolSizePerDevice,
 	return DpdkDeviceList::getInstance().initDpdkDevices(m_MBufPoolSizePerDevice);
 }
 
+bool DpdkDeviceList::externalInitializationDpdk(CoreMask coreMask, uint32_t mBufPoolSizePerDevice, uint8_t masterCore)
+{
+    m_CoreMask = coreMask;
+	m_IsDpdkInitialized = true;
+    m_MBufPoolSizePerDevice = mBufPoolSizePerDevice;
+	DpdkDeviceList::getInstance().setDpdkLogLevel(Logger::Info);
+	return DpdkDeviceList::getInstance().initDpdkDevices(m_MBufPoolSizePerDevice);
+}
+
 bool DpdkDeviceList::initDpdkDevices(uint32_t mBufPoolSizePerDevice)
 {
 	if (!m_IsDpdkInitialized)
